extensions/signature.md

```diff
@@ -0,0 +1,108 @@
+# Signature Extension
+
+This extension defines a standard mechanism to carry signatures allowing
+event consumers to ensure the identity of the event producer and verify
+the event-data has not been tampered with during transport.
+
+This specification does not:
+
+- Declare a mechanism for public key exchange.
+- Declare a mechanism for public key selection during signature verification.
+Implementors may consider leveraging the CloudEvent `source` as a method
+of selection.
+
+## Usage
+
+This extension supports a few general usage models:
+
+- Where the algorithm used to produce an attached signature needs to be shared.
+- Where the `data` is signed with a detached signature.
+- Where the CloudEvent context attributes are signed.
+
+## Attributes
+
+### sigdata
+
+- Type: `Binary`
+- Description: The binary signature generated for the event `data`.
+- Constraints:
+
+  - OPTIONAL
+
+### sigdataalgo
+
+- Type : `String`
+- Description : The algorthim used to generate the `data` signature.
```

Comment:
Given the size constraints on each event, can there be a set of algorithms that are strongly recommended by this specification?

Reply:
I'm loathe to get down to that level of detail, I see CE's as containers to enable data interchange, not to specification the "how" - does that make sense?